pkg/sql/sem/builtins/builtins.go

```diff
@@ -2595,6 +2595,113 @@ var builtins = map[string]builtinDefinition{
 		},
 	),
 
+	// https://www.postgresql.org/docs/14/functions-datetime.html#FUNCTIONS-DATETIME-TABLE
+	//
+	// PostgreSQL documents date_trunc for text and double precision.
+	// It will also handle smallint, integer, bigint, decimal,
+	// numeric, real, and numeri like text inputs by casting them,
+	// so we support those for compatibility. This gives us the following
+	// function signatures:
+	//
+	//  to_timestamp(text, text)       -> TimestampTZ
+	//  to_timestamp(text)             -> TimestampTZ
+	//  to_timestamp(INT)              -> TimestampTZ
+	//  to_timestamp(INT2)             -> TimestampTZ
+	//  to_timestamp(INT4)             -> TimestampTZ
+	//  to_timestamp(INT8)             -> TimestampTZ
+	//  to_timestamp(FLOAT)            -> TimestampTZ
+	//  to_timestamp(REAL)             -> TimestampTZ
+	//  to_timestamp(DOUBLE PRECISION) -> TimestampTZ
+	//  to_timestamp(DECIMAL)          -> TimestampTZ
+	//
+	// See the following snippet from running the functions in PostgreSQL:
+	//
+	//               postgres=# select to_timestamp(32767::smallint);
+	//               to_timestamp
+	//               ------------------------
+	//               1970-01-01 09:06:07+00
+	//
+	//               postgres=# select to_timestamp(1646906263::integer);
+	//               to_timestamp
+	//               ------------------------
+	//               2022-03-10 09:57:43+00
+	//
+	//               postgres=# select to_timestamp(1646906263::bigint);
+	//               to_timestamp
+	//               ------------------------
+	//               2022-03-10 09:57:43+00
+	//
+	//               postgres=# select to_timestamp(1646906263.123456::decimal);
+	//               to_timestamp
+	//               -------------------------------
+	//               2022-03-10 09:57:43.123456+00
+	//
+	//               postgres=# select to_timestamp(1646906263.123456::numeric);
+	//               to_timestamp
+	//               -------------------------------
+	//               2022-03-10 09:57:43.123456+00
+	//
+	//               postgres=# select to_timestamp(1646906263.123456::real);
+	//               to_timestamp
+	//               ------------------------
+	//               2022-03-10 09:57:20+00
+	//
+	//               postgres=# select to_timestamp('1646906263.123456');
+	//               to_timestamp
+	//               -------------------------------
+	//               2022-03-10 09:57:43.123456+00
+	//
+	"to_timestamp": makeBuiltin(
+		tree.FunctionProperties{Category: categoryDateAndTime},
+		tree.Overload{
+			Types:      tree.ArgTypes{{"timestamp", types.String}},
```

Comment:
It's similar, but `experimental_strptime` doesn't support the same formatting strings as `to_timestamp`. Here's an example from the Postgres docs:

```
defaultdb> select experimental_strptime('05 Dec 2000', 'DD Mon YYYY');
  experimental_strptime
--------------------------
  0001-01-01 00:00:00+00
```

The result should be `2000-12-05 00:00:00-05` (assuming the timezone is set to `-05`). It looks like `experimental_strptime` returns a default value when parsing fails, rather than erring.